ingest/src/main/java/com/microsoft/azure/kusto/ingest/IngestClientImpl.java

```diff
@@ -204,8 +222,124 @@ private String genBlobName(String fileName, String databaseName, String tableNam
+            FileSourceInfo fileSourceInfo = new FileSourceInfo(tempFile.getAbsolutePath(), numberOfChars * 2);
+            IngestionResult ingestionResult = ingestFromFile(fileSourceInfo, ingestionProperties);
+
+            //noinspection ResultOfMethodCallIgnored
+            tempFile.delete();
+
+            return ingestionResult;
+        } catch (IngestionClientException | IngestionServiceException ex) {
+            log.error("Unexpected error when ingesting a result set.", ex);
+            throw ex;
+        } catch (IOException ex) {
+            String msg = "Failed to write or delete local file";
+            log.error(msg, ex);
+            throw new IngestionClientException(msg);
+        }
+    }
+
+    long resultSetToCsv(ResultSet resultSet, Writer writer, boolean includeHeaderAsFirstRow) throws IngestionClientException {
+        final String LINE_SEPARATOR = System.getProperty("line.separator");
+
+        try {
+            String columnSeparator = "";
+
+            ResultSetMetaData metaData = resultSet.getMetaData();
+            int numberOfColumns = metaData.getColumnCount();
+
+            if (includeHeaderAsFirstRow) {
+                for (int column = 0; column < numberOfColumns; column++) {
+                    writer.write(columnSeparator);
+                    writer.write(metaData.getColumnLabel(column + 1));
+
+                    columnSeparator = ",";
+                }
+
+                writer.write(LINE_SEPARATOR);
+            }
+
+            int numberOfRecords = 0;
+            long numberOfChars = 0;
+
+            // Get all rows.
+            while (resultSet.next()) {
+                numberOfChars += writeResultSetRow(resultSet, writer, numberOfColumns);
+                writer.write(LINE_SEPARATOR);
+                // Increment row count
+                numberOfRecords++;
+            }
+
+            log.debug("Number of chars written from column values: {}", numberOfChars);
+
+            long totalNumberOfChars = totalNumberOfCharsWrittenToCsv(numberOfChars, numberOfColumns
+                    , numberOfRecords, LINE_SEPARATOR.length());
+
+            log.debug("Wrote resultset to file. CharsCount: {}, ColumnCount: {}, RecordCount: {}"
+                    , numberOfChars, numberOfColumns, numberOfRecords);
+
+            return totalNumberOfChars;
+        } catch (Exception ex) {
+            String msg = "Unexpected error when writing result set to temporary file.";
+            log.error(msg, ex);
+            throw new IngestionClientException(msg);
+        } finally {
+            try {
+                writer.close();
+            } catch (IOException e) { /* ignore */
+            }
+        }
+    }
+
+    private int writeResultSetRow(ResultSet resultSet, Writer writer, int numberOfColumns) throws IOException, SQLException {
+        int numberOfChars = 0;
+        String columnString;
+        String columnSeparator = "";
+
+        for (int i = 1; i <= numberOfColumns; i++) {
```

Comment:
Why 1? what do we have in the first column?

Reply:
JDBC ResultSet is not a 0 based, so column 0 is always empty